### PR TITLE
Enhance 'install-powershell.ps1' to work on Linux/MacOS

### PR DESCRIPTION
**Pull request by daxian-dbw**
Enhance 'install-powershell.ps1' to work on Linux/MacOS.

When `-AddToPath` is specified:
- On Windows, add the absolute destination path to the 'User' scope environment variable 'Path';
- On Linux, make the symlink '/usr/bin/pwsh' points to "$Destination/pwsh";
- On MacOS, make the symlink '/usr/local/bin/pwsh' points to "$Destination/pwsh".

### CODE REVIEW

**Review by TravisEz13 — APPROVED** [review]

File: tools/install-powershell.ps1 [hunk 1]
```diff
@@ -85,25 +107,75 @@ try {
...
-        ## Add to the 'Path' for the current process
-        $env:Path = $Destination + [System.IO.Path]::PathSeparator + $env:Path
-        Write-Verbose "'$Destination' is added to Path" -Verbose
+    if ($AddToPath) {
+        if ($IsWinEnv -and (-not $env:Path.Contains($Destination))) {
+            ## Add to the User scope 'Path' environment variable
+            $userPath = [System.Environment]::GetEnvironmentVariable("Path", "User")
+            $userPath = $Destination + [System.IO.Path]::PathSeparator + $userPath
+            [System.Environment]::SetEnvironmentVariable("Path", $userPath, "User")
+            Write-Verbose "'$Destination' is added to the Path" -Verbose
```
Comment (adityapatwardhan) [inline tools/install-powershell.ps1]: I think, since we have CmdletBinding() we should not force -verbose

File: tools/install-powershell.ps1 [hunk 2]
```diff
@@ -85,25 +107,75 @@ try {
...
+
         $downloadURL = "https://github.com/PowerShell/PowerShell/releases/download/v${release}/${packageName}"
         Write-Verbose "About to download package from '$downloadURL'" -Verbose
 
         $packagePath = Join-Path -Path $tempDir -ChildPath $packageName
         Invoke-WebRequest -Uri $downloadURL -OutFile $packagePath
 
-        Expand-Archive -Path $packagePath -DestinationPath $Destination
+        if ($IsWinEnv) {
+            Expand-Archive -Path $packagePath -DestinationPath $Destination
```
Comment (adityapatwardhan) [inline tools/install-powershell.ps1]: Should we delete the compressed package after expansion?

**Review by adityapatwardhan — APPROVED** [review]
Some non-blocking comments

File: tools/install-powershell.ps1 (hunk 2 above)
Comment (daxian-dbw) [reply tools/install-powershell.ps1]: The compressed package will be deleted. The compressed package is downloaded to `$tempDir` which will be deleted in the finally block at the end of the script.

File: tools/install-powershell.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply tools/install-powershell.ps1]: These messages are actually intentionally written out so that a user roughly knows what step the script is doing.

**Comment by daxian-dbw** [comment]
Linux job succeeded, but the macOS job was canceled again.
Currently `install-powershell.ps1` is only used in AppVeyor bootstrapping (install latest powershell core if not found), so it won't affect macOS CI run. Given that, I will merge this PR.

**Pull request merged**